src/obr/cli.py

```diff
@@ -292,6 +308,148 @@ def query(ctx: click.Context, **kwargs):
     query_impl(project, queries, output=True, latest_only=not kwargs.get("all"))
 
 
+@cli.command()
+@click.option(
+    "--filter",
+    type=str,
+    multiple=True,
+    help=(
+        "Pass a <key>=<value> value pair per occurrence of --filter. For instance, obr"
+        " archive --filter solver=pisoFoam --filter preconditioner=IC"
+    ),
+)
+@click.option(
+    "-f",
+    "--folder",
+    required=True,
+    default=".",
+    type=str,
+    help="Path to OpenFOAMProject.",
+)
+@click.option(
+    "-r",
+    "--repo",
+    required=True,
+    type=str,
+    help=(
+        "Path to data repository. If this is a valid Github repository, files will be"
+        " automatically added."
+    ),
+)
+@click.option(
+    "-s",
+    "--skip-logs",
+    required=False,
+    is_flag=True,
+    help=(
+        "If set, .log files will not be archived. This does not affect .log files"
+        " passed via the --file option."
+    ),
+)
+@click.option(
+    "-a",
+    "--file",
+    required=False,
+    multiple=True,
+    help="Path(s) to non-logfile(s) to be also added to the repository.",
+)
+@click.pass_context
+def archive(ctx: click.Context, **kwargs):
+    if current_path := kwargs.get("folder", "."):
+        os.chdir(current_path)
+        current_path = Path(current_path).absolute()
+
+    # setup project and jobs
+    project = OpenFOAMProject().init_project()
+    filters: tuple[str] = kwargs.get("filter", ())
+    jobs = filter_jobs_by_query(project, filters)
+
+    time = str(datetime.now()).replace(" ", "_")
+    target_folder: str = kwargs.get("repo", "")
+    use_github_repo = False
+    repo = None
+    branch_name = None
+    previous_branch = None
+    # check if given path is actually a github repository
+    try:
+        repo = Repo(path=target_folder)
+        previous_branch = repo.active_branch.name
+        branch_name = "archive-" + (
+            str(datetime.now()).rsplit(":", 1)[0].replace(" ", ":").replace(":", "_")
+        )
+        use_github_repo = True
+        logging.info(f"checkout archive-{branch_name}")
+        repo.git.checkout("HEAD", b=branch_name)
+    except InvalidGitRepositoryError:
+        logging.warn(
+            f"Given directory {target_folder=} is not a github repository. Will only"
+            " copy files."
+        )
+
+    # setup target folder
+    path = Path(target_folder + f"/{time}")
```

Comment:
This is a reasonable idea to store by timestamp, however, I think it causes problems because we want to be able to add logs to the archive at a later point and to post-process all logs with obr once we have a collection. Also, at least when archiving log files, the logs have a time stamp and won't be overridden. Other result files might need a different approach.